packages/block-library/src/more/edit.js

```diff
@@ -3,82 +3,68 @@
  */
 import { __ } from '@wordpress/i18n';
 import { PanelBody, ToggleControl } from '@wordpress/components';
-import { Component } from '@wordpress/element';
 import { InspectorControls } from '@wordpress/block-editor';
 import { ENTER } from '@wordpress/keycodes';
 import { getDefaultBlockName, createBlock } from '@wordpress/blocks';
 
-export default class MoreEdit extends Component {
-	constructor() {
-		super( ...arguments );
-		this.onChangeInput = this.onChangeInput.bind( this );
-		this.onKeyDown = this.onKeyDown.bind( this );
+const DEFAULT_TEXT = __( 'Read more' );
 
-		this.state = {
-			defaultText: __( 'Read more' ),
-		};
-	}
+export default function MoreEdit( {
+	// Before refactoring we allowed saving empty customText with an `undefined` value
+	// but we misleadingly were showing the DEFAULT_TEXT.
+	// So for backwards compatibility we set the same Default value to custom text.
```

Comment:
Some slight improvements to the comment:
```suggestion
	// Before refactoring we allowed saving empty customText with an `undefined` value,
	// but we misleadingly displayed the DEFAULT_TEXT in the editor.
	// So for backwards compatibility, we default customText to the same value.
```

But I don't think we should be setting the `customText` default to `DEFAULT_TEXT` in the first place. Some themes have different default text for More tags; Twenty Twenty uses "continue reading", for example. (Note that this default text never appears in the editor.) If `customText` is given a non-empty value upon insertion, then the More block will always be overriding the theme default unless the user explicitly empties the `customText` field.

It is kinda broken to display `DEFAULT_TEXT` in the editor when the front-end output is different, but it's just as broken to override the theme default by default, in my opinion.

A proper solution would be to use something like a placeholder in the more tag text field, rather than pre-populating it with an actual value. In other words, make it act more like the Heading block, which displays "Write heading..." by default.

If you want, I could push a commit to do that.

Reply:
Alternatively, I guess you could revert the behavior to act the same as `master` and I could make a separate PR to properly fix this issue separately. That's probably the best way to handle this.

Reply:
As @mcsf also noted above:
>The fact that — prior to this PR — the MoreEdit component already had defaultText: __( 'Read more' ) is probably a mistake.
>I think this PR should just be about the refactoring. :)

So I will probably keep the previous functionality to allow the empty text.

Reply:
> Alternatively, I guess you could revert the behavior to act the same as master

Hey @ZebulanStanphill. Thanks for taking the time looking into this -- since I'll revert the `default` value, can you unblock the PR from the requested changes in the comments?

Thanks!